File: src/spacerocks/calc_M_from_E.cpp
#include "spacerocks.hpp"

double calc_M_from_E(double e, double E) {

  double M;


  if (e < 1) {
    M = E - e * sin(E);
  }
  else {
    M = e * sinh(E) - E;
  }

  return M;

}

File: src/transforms/calc_M_from_E.rs
#[allow(non_snake_case)]
pub fn calc_M_from_E(e: f64, E: f64) -> f64 {
    let mut M;

    if e < 1.0 {
        M = E - e * E.sin();
    }

    else {
        M = e * E.sinh() - E;
    }

    return M;
}